presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -163,18 +163,28 @@ default int compareTo(int leftPosition, int leftOffset, int leftLength, Block ri
     int getPositionCount();
 
     /**
-     * Returns the logical size of this block in memory.
+     * Returns the size of this block as if it was compacted, ignoring any over-allocations.
      */
     long getSizeInBytes();
 
     /**
-     * Returns the logical size of {@code block.getRegion(position, length)} in memory.
+     * Returns the size of this block's contents as if no optimizations were applied to it.
+     * As example, for RLE and Dictionary blocks this can differ significantly from
+     * {@code block.getSizeInBytes()}.
```

Comment:
I tried to expand on this. What do you think?
```
This can differ substantially from {@link #getSizeInBytes} for dictionary
or RLE blocks. For RLE, it will be {@code N} times larger. For dictionary,
it will be larger based on how many times dictionary entries reused.

Reply:
Looks good. Will update the comment.